test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -29,34 +35,80 @@
 #include <grpcpp/server.h>
 #include <grpcpp/server_builder.h>
 
+#include "src/core/lib/gprpp/cpp_impl_of.h"
+#include "src/core/tsi/ssl_transport_security.h"
+#include "src/cpp/client/secure_credentials.h"
 #include "test/core/util/port.h"
 #include "test/core/util/test_config.h"
 #include "test/core/util/tls_utils.h"
 #include "test/cpp/end2end/test_service_impl.h"
 
+// TODO(matthewstevenson88): More test cases to add:
+// - Use P256, P384, P512 credentials.
+// - Use a long certificate chain.
+// - Use a large certificate.
+// - Large trust bundle.
+// - Bad ALPN.
+// - In TLS 1.2, play with the ciphersuites.
+// - More failure modes.
+// - Certs containing more SANs.
+// - Copy all of this over to tls_credentials_test.cc.
+// - Client doesn't have cert but server requests one.
+// - Bad session ticket in cache.
+// - Use same channel creds object on sequential/concurrent handshakes.
+// - Do successful handshake with a localhost server cert.
+// - Missing or malformed roots on both sides.
+
 namespace grpc {
 namespace testing {
 namespace {
 
+using ::grpc_core::testing::GetFileContents;
+using ::testing::HasSubstr;
+using ::testing::IsEmpty;
+using ::testing::UnorderedElementsAre;
+
 constexpr char kCaCertPath[] = "src/core/tsi/test_creds/ca.pem";
 constexpr char kServerCertPath[] = "src/core/tsi/test_creds/server1.pem";
 constexpr char kServerKeyPath[] = "src/core/tsi/test_creds/server1.key";
 constexpr char kClientCertPath[] = "src/core/tsi/test_creds/client.pem";
 constexpr char kClientKeyPath[] = "src/core/tsi/test_creds/client.key";
 constexpr char kMessage[] = "Hello";
 
-class SslCredentialsTest : public ::testing::Test {
+struct ToStringJoinFormatter {
```

Comment:
Doesn't look like this is used anywhere.

Reply:
Good catch, removed.